lms/static/scripts/frontend_apps/services/client-rpc.js

```diff
@@ -11,6 +13,9 @@ import { JWT } from '../utils/jwt';
 /**
  * @typedef ServiceConfig
  * @prop {string} grantToken
+ * @prop {string} notifyOnAnnotationActivity - A string identifying an RPC
+ *   method the client should call when qualifying annotation activity occurs
+ *   (save, edit, e.g.). Takes the form `'$rpc:<methodName>'`
```

Comment:
I copied the syntax of the `groups` configuration value which uses the `$rpc` prefix.

Reply:
The `$rpc:` prefix is used with the `groups` property to distinguish it from a fixed group ID. The `requestGroups` method name isn't actually configurable either in that context.

I would be inclined to generalize this slightly in case we want to add other events in future or a different reporting method. 

```js
reportActivity: {
  // How to report activity. Currently there is only one supported method, so we could leave this out.
  method: 'rpc',
  // List of events to report.
  events: [...],
}
```

Note there is no configuration for the event name here. I can't see a reason yet why that should be configurable.

Reply:
@robertknight  I want to mention now before I forget that making any RPC calls from the client to the LMS app is currently coupled with these URL parameters provided by Via:

```
requestConfigFromFrame: {
  ancestorLevel: 2, // How many levels "up" the parent frame is from the sidebar frame
  origin: 'http://localhost:8001', // RPC origin
}
```

That means that presently "reportActivity" support would only be possible for sidebar instances that use RPC to get their settings, if that makes any sense.

In future, it might be as simple as restructuring/renaming those Via-provided configuration values perhaps something like:

```
requestConfigFromFrame: {
  method: 'rpc:methodName',
},
rpc: {
  ancestorLevel: 2,
  origin: 'http://localhost:8001',
}
```

Or somesuch. Point being that the `ancestorLevel` and `origin` values are more widely applicable than just the config request.